SourceCode/Scene/Textures/Texture.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include <string>
+#include "../../Utils/CRTVector.h"
+
+struct Texture {
+    std::string name;
+
+    Texture(const std::string& name) : name(name) {}
+
+    virtual CRTVector sample(float u, float v) const = 0;
```

Comment:
Nice idea, but we have to be careful with the dynamic polymorphism it could introduce undesired overhead, at a later stage it would be good idea to profile with and without it.